lib/structures/GuildScheduledEvent.js

```diff
@@ -0,0 +1,154 @@
+"use strict";
+
+const Base = require("./Base");
+const Endpoints = require("../rest/Endpoints");
+
+/**
+* Represents a guild scheduled event
+* @prop {(VoiceChannel | StageChannel | Object)?} channel The channel where the event will be held. This will be null if the event is external (`entityType` is `3`). Can be partial with only `id` if the channel or guild is not cached
+* @prop {User?} creator The user that created the scheduled event. For events created before October 25 2021, this will be null. Please see the relevant Discord documentation for more details
+* @prop {String?} description The description of the event
+* @prop {String?} entityID The entity ID associated to the event
+* @prop {Object?} entityMetadata Metadata for the event. This will be null if the event is not external (`entityType` is not `3`)
+* @prop {String?} entityMetadata.location Location of the event
+* @prop {Number} entityType The [entity type](https://discord.com/developers/docs/resources/guild-scheduled-event#guild-scheduled-event-object-guild-scheduled-event-entity-types) of the scheduled event
+* @prop {Guild | Object} guild The guild which the event belongs to. Can be partial with only `id` if not cached
+* @prop {String} id The id of the guild event
+* @prop {String?} image The hash of the event's image, or null if no image
+* @prop {String?} imageURL The URL of the event's image, or null if no image
+* @prop {String} name The name of the event
+* @prop {Number} privacyLevel Event privacy level
+* @prop {Number} scheduledStartTime The time the event will start
+* @prop {Number?} scheduledEndTime The time the event will end, or null if the event does not have a scheduled time to end
+* @prop {Number} status The [status](https://discord.com/developers/docs/resources/guild-scheduled-event#guild-scheduled-event-object-guild-scheduled-event-status) of the scheduled event
+* @prop {Number?} userCount The number of users subscribed to the event
+*/
+class GuildScheduledEvent extends Base {
+    constructor(data, client) {
+        super(data.id);
+
+        this._client = client;
+        if(data.creator !== undefined) {
+            this.creator = client.users.update(data.creator, this.client);
+        } else {
+            this.creator = null;
+        }
+        this.guild = client.guilds.get(data.guild_id) || {id: data.guild_id};
```

Comment:
```suggestion
        this.guild = client.guilds.get(data.guild_id) || {
            id: data.guild_id
        };
```

Use the same formatting as other files.